mcux/hwmv1.cmake

```diff
@@ -0,0 +1,46 @@
+# Get MCUX_DEVICE and MCUX_DEVICE_PATH from CONFIG_SOC
+# For example, if CONFIG_SOC=mimx8ml8_ca53, then MCUX_DEVICE=MIMX8ML8_ca53
+# and MCUX_DEVICE_PATH=MIMX8ML8
+string(FIND ${CONFIG_SOC} "_" pos)
+if(pos GREATER -1)
+  string(REPLACE "_" ";" MCUX_DEVICE_TMP ${CONFIG_SOC})
+  list(GET MCUX_DEVICE_TMP 0 MCUX_DEVICE_SOC)
+  list(GET MCUX_DEVICE_TMP 1 MCUX_DEVICE_CORE)
+  string(TOUPPER ${MCUX_DEVICE_SOC} MCUX_DEVICE_PATH)
+  set(MCUX_DEVICE ${MCUX_DEVICE_PATH}_${MCUX_DEVICE_CORE})
```

Comment:
we still need these code for hardware model v2, in fact these code can achive what we needed for v2. 
For example, if CONFIG_SOC=mimx8ml8_ca53, then MCUX_DEVICE=MIMX8ML8_ca53 and MCUX_DEVICE_PATH=MIMX8ML8, and MCUX_CPU = CPU_${CONFIG_SOC_PART_NUMBER})
In fact, I still don't clear what's the difference for v2 and v1 you want to update in this PR.

Reply:
I found Cortex-A Core's Part number are with cpu suffix, such as CONFIG_SOC_PART_NUMBER="MIMX8MM6DVTLZ_ca53"
So I will update them to remove cpu suffix. maybe this is one difference for hardware model v1 and v2.

Reply:
For hardware model V2, the `CONFIG_SOC` value should match the SOC name given in `soc.yml`, which will be `mimx8ml8` for the case you mention. Note that this restriction is being enforced by a CI check upstream (https://github.com/zephyrproject-rtos/zephyr/pull/69309), which is why we are trying to align with it here. Once we remove CPU suffixes, we can use `CONFIG_SOC_xxx_CORE_NAME` Kconfigs, like `CONFIG_SOC_MIMX8ML8_CA53` to set the `MCUX_CORE_SUFFIX` variable

In your case, I would expect the following Kconfig values:
```
CONFIG_SOC_PART_NUMBER=MIMX8MM6DVTLZ
CONFIG_SOC=MIMX8ML8
```

Generally, we will align `CONFIG_SOC` with the folder name used for the SOC in MCUX SDK